src/components/FilterDropdown.vue

```diff
@@ -0,0 +1,478 @@
+                <a href="#" @click="reset">
+                    <span class="ui text" :class="{ [color]: !!color }">Reset</span>
+                </a>
+            </div>
+
+            <!-- Search container -->
+            <div v-if="!hideSearch" class="s-container mb-4">
+                <div class="ui fluid icon small input">
+                    <input v-model="searchKey" type="text" placeholder="Search" />
+                    <i v-if="searchKey" class="times thin icon link" @click="searchKey = ''" />
+                </div>
+                <!-- No data text -->
+                <span v-if="listItems.length === 0" class="ui text grey mt-4 d-flex justify-content-center">{{
+                    noDataText
+                }}</span>
+            </div>
+
+            <!-- Prepend list slot -->
+            <div v-if="hasPrependListSlot">
+                <slot name="prepend-list" />
+            </div>
+
+            <!-- Items -->
+            <div
+                v-if="listItems.length > 0"
+                class="ui grid fluid vertical menu sub-menu"
+                :style="{ maxHeight: parsePxStyle(scrollHeight) }"
+            >
+                <div
+                    v-for="(item, index) in listItems"
+                    :key="index"
+                    :class="[item.type, { disabled: item.disabled, active: isSelected(item) }]"
+                    @click.stop="item.type.includes('item') ? handleItemClick(item) : undefined"
+                >
+                    <!-- if this is an item -->
+                    <template v-if="item.type.includes('item')">
+                        <div class="ui checkbox" :class="{ [size]: !!size }">
+                            <input type="checkbox" :name="item.value" :checked="isSelected(item)" />
+                            <label>{{ item.text }}</label>
+                        </div>
+                        <div v-if="item.description" class="description">{{ item.description }}</div>
+                    </template>
+                    <!-- not an item, put as plain text -->
+                    <template v-else>{{ item.text }}</template>
+                </div>
+            </div>
+
+            <!-- Append list slot -->
+            <div v-if="hasAppendListSlot">
+                <slot name="append-list" />
+            </div>
+        </div>
+    </div>
+</template>
+<script>
+import { computed, reactive, toRefs, onMounted, watch } from '@vue/composition-api'
+import { parsePxStyle, validateSize } from './mixins'
+import { search } from '../utils'
+
+export default {
+    props: {
+        value: {
+            type: Array,
+            required: true
+        },
+        title: String,
+        items: {
+            type: Array,
+            required: true
+        },
+        itemText: {
+            default: 'text'
+        },
+        itemValue: {
+            default: 'value'
+        },
+        returnType: {
+            type: String,
+            default: 'value'
```

Comment:
can we have a validation for acceptable types here?

Reply:
The thing with returnType is, the field user defined for value can be different than the default.

So lets say user use `item-value="abc"` instead of `value` so if we put a validator for `returnType` and only accept `value` or `object` as a `returnType` then the user cannot define `abc` as `returnType` so it must to be dynamic